src/manual_inspection.py

```diff
@@ -0,0 +1,183 @@
+import cv2
+import hydra
+import getpass
+import logging
+import datetime
+import pandas as pd
+from pathlib import Path
+from omegaconf import DictConfig
+import time
+
+# Configure logging
+log = logging.getLogger(__name__)
+
+GITHUB_REPO_URL = "https://github.com/precision-sustainable-ag/Field-AnnotationPipeline/issues"
+
+LABEL_OPTIONS = {
+                "1": "Good Mask",
+                "2": "Bad Mask",
+                "3": "Incorrect Species",
+                "0": "Other",
+                "q": "Quit"
+            }
+
+class ManualInspection:
+    """
+    Class to manually inspect images and label them based on quality.
+    """
+    def __init__(self, cfg: DictConfig):
+        self.batch_id = cfg.batch_id
+        self.longterm_inspection_dir = Path(cfg.paths.longterm_storage) / "field-batches"/ self.batch_id/ "inspection"
+        self.csv_file = self.longterm_inspection_dir / f"{self.batch_id}_preprocessing_inspection_results.csv"
+        self.images = self._load_images()
+        self.results = self._load_existing_results()
+        self.timestamp = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        self.user = getpass.getuser()
+
+    def _load_images(self):
+        """Load images and return a sorted list of unlabeled ones."""
+        all_images = sorted(self.longterm_inspection_dir.glob("*.jpg"))
+        if not all_images:
+            log.warning(f"No images found in {self.longterm_inspection_dir}")
+            return []
+
+        return [img for img in all_images if img.stem not in self._get_labeled_images()]
+
+    def _get_labeled_images(self):
+        """Retrieve a set of already labeled images from the CSV file."""
+        if self.csv_file.exists():
+            df_existing = pd.read_csv(self.csv_file)
+            return set(df_existing['ImageID'].tolist())
+        return set()
+
+    def _load_existing_results(self):
+        """Load existing CSV results or return an empty list."""
+        if self.csv_file.exists():
+            log.info(f"Loading existing results from {self.csv_file}")
+            return pd.read_csv(self.csv_file).values.tolist()
+        return []
+
+    def _display_instructions(self):
+        """Prints instructions for user input."""
+        print("\n--- Segmentation Quality Assessment ---")
+        print_labels = {
```

Comment:
If you're not using emojis, just use the global LABEL_OPTIONS

Reply:
I added emojis.